{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import subprocess\n",
    "import re\n",
    "from __future__ import print_function\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "file_bam = '/ref/Cre/genemodel_correction/tophat/SRR2132433.tophat_out_maxintron1000.accepted_hits.bam'\n",
    "#df_gene_gff = pd.read_pickle('../../genemodel_correction/tophat/df.cre.gff.gene.index.pickle')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "Preparing GFF file into df_indexed. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python2.7/dist-packages/ipykernel/__main__.py:30: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/usr/local/lib/python2.7/dist-packages/ipykernel/__main__.py:31: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>transcriptname</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genename</th>\n",
       "      <th>longest</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>gene</td>\n",
       "      <td>29681</td>\n",
       "      <td>31614</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=311229.v1;Name=311229</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">311229.v1</th>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>29681</td>\n",
       "      <td>31614</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=311229.v1.107;Name=311229;pacid=16057706;lo...</td>\n",
       "      <td>311229.v1.107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>exon</td>\n",
       "      <td>29681</td>\n",
       "      <td>29746</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=311229.v1.107.exon.1;Parent=311229.v1.107;p...</td>\n",
       "      <td>311229.v1.107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>CDS</td>\n",
       "      <td>29681</td>\n",
       "      <td>29746</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=311229.v1.107.CDS.1;Parent=311229.v1.107;pa...</td>\n",
       "      <td>311229.v1.107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>exon</td>\n",
       "      <td>29864</td>\n",
       "      <td>29938</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=311229.v1.107.exon.2;Parent=311229.v1.107;p...</td>\n",
       "      <td>311229.v1.107</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            0             1     2      3      4  5  6  7  \\\n",
       "genename  longest                                                          \n",
       "NaN       NaN      scaffold_1  phytozomev10  gene  29681  31614  .  +  .   \n",
       "311229.v1 1        scaffold_1  phytozomev10  mRNA  29681  31614  .  +  .   \n",
       "          1        scaffold_1  phytozomev10  exon  29681  29746  .  +  .   \n",
       "          1        scaffold_1  phytozomev10   CDS  29681  29746  .  +  0   \n",
       "          1        scaffold_1  phytozomev10  exon  29864  29938  .  +  .   \n",
       "\n",
       "                                                                   8  \\\n",
       "genename  longest                                                      \n",
       "NaN       NaN                               ID=311229.v1;Name=311229   \n",
       "311229.v1 1        ID=311229.v1.107;Name=311229;pacid=16057706;lo...   \n",
       "          1        ID=311229.v1.107.exon.1;Parent=311229.v1.107;p...   \n",
       "          1        ID=311229.v1.107.CDS.1;Parent=311229.v1.107;pa...   \n",
       "          1        ID=311229.v1.107.exon.2;Parent=311229.v1.107;p...   \n",
       "\n",
       "                  transcriptname  \n",
       "genename  longest                 \n",
       "NaN       NaN               None  \n",
       "311229.v1 1        311229.v1.107  \n",
       "          1        311229.v1.107  \n",
       "          1        311229.v1.107  \n",
       "          1        311229.v1.107  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_gff = '/ref/Cre/DroughtNet/PhytozomeV10_download/References/Alyrata/annotation/Alyrata_107_v1.0.gene_exons.gff3'\n",
    "df_gff   = pd.read_csv(file_gff,sep='\\t',skiprows=2,header=None)\n",
    "\n",
    "# For CRE\n",
    "#df_gff['genename'] = df_gff[8].apply(lambda x : '.'.join(x.split(';')[0].replace('ID=','').split('.')[0:2]))\n",
    "#df_gff['transcriptname'] = df_gff[8].apply(lambda x : '.'.join(x.split(';')[0].replace('ID=','').split('.')[0:4]))\n",
    "\n",
    "\n",
    "dicT2G      = {} # transcript to genename\n",
    "df_gff_mRNA = df_gff[(df_gff[2] == 'mRNA')]\n",
    "def get_genename(x):\n",
    "    if x[2] == 'gene':\n",
    "        return None\n",
    "    key   = [y.split('=')[0] for y in x[8].split(';')]\n",
    "    value = [y.split('=')[1] for y in x[8].split(';')]\n",
    "    dic   = dict(zip(key,value))\n",
    "    return dic['Parent']\n",
    "def get_transcript(x):\n",
    "    \n",
    "    \n",
    "    key   = [y.split('=')[0] for y in x[8].split(';')]\n",
    "    value = [y.split('=')[1] for y in x[8].split(';')]\n",
    "    dic   = dict(zip(key,value))\n",
    "    if x[2] == 'gene':\n",
    "        return None\n",
    "    elif x[2] == 'mRNA':\n",
    "        return dic['ID']\n",
    "    else: \n",
    "        return dic['Parent']\n",
    "df_gff_mRNA['genename']   = df_gff_mRNA.apply(get_genename,axis=1)\n",
    "df_gff_mRNA['transcript'] = df_gff_mRNA.apply(get_transcript,axis=1)\n",
    "df_gff_mRNA_ix = df_gff_mRNA.set_index('transcript')\n",
    "# For ESA\n",
    "df_gff['transcriptname']  = df_gff.apply(get_transcript,axis=1)\n",
    "\n",
    "\n",
    "\n",
    "def get_genename2(x):\n",
    "    if x != None:\n",
    "        return df_gff_mRNA_ix.loc[x]['genename']\n",
    "    else:\n",
    "        return None\n",
    "df_gff['genename'] = df_gff['transcriptname'].apply(get_genename2)\n",
    "\n",
    "\n",
    "\n",
    "# grep longest transcript names\n",
    "df_gff_mRNA_ix['longest'] = df_gff_mRNA_ix[8].apply(lambda x : x.split(';')[3].replace('longest=',''))\n",
    "#df_gff_mRNA_index =  df_gff_mRNA.set_index('transcript')\n",
    "\n",
    "def get_longest(x):\n",
    "    if x == None:\n",
    "        return None\n",
    "    try:\n",
    "        return df_gff_mRNA_ix.loc[x]['longest']\n",
    "    except KeyError:\n",
    "        return None\n",
    "df_gff['longest'] = df_gff['transcriptname'].apply(get_longest)\n",
    "\n",
    "df_gff_index = df_gff.set_index(['genename','longest'])\n",
    "\n",
    "df_gff_index.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_gff_index.to_pickle(file_gff+'.pandas.df.pk')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_gff_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_gff_index.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "genename = 'Cre15.g635350'\n",
    "chromosome, left, right = df_gene_gff.loc[genename][[0,3,4]]\n",
    "\n",
    "print ('samtools view %s %s:%d-%d > temp.sam'%(file_bam,chromosome,left,right))\n",
    "subprocess.call('samtools view %s %s:%d-%d > temp.sam'%(file_bam,chromosome,left,right),shell=True)\n",
    "subprocess.call('cut -f 1-20 temp.sam > temp.sam.cut',shell=True)\n",
    "\n",
    "df_sam = pd.read_csv('temp.sam.cut',sep='\\t',header=None)\n",
    "\n",
    "df_sam['SRRID']  = df_sam[0].apply(lambda x:x.split('.')[0])\n",
    "df_sam['READID'] = df_sam[0].apply(lambda x:x.split('.')[1])\n",
    "df_sam['PAIRID'] = df_sam[0].apply(lambda x:x.split('.')[2])\n",
    "\n",
    "mask = (df_sam[8] > 0)\n",
    "df_sam_forward = df_sam[mask]\n",
    "df_sam_forward = df_sam_forward.sort(3)\n",
    "READID_LIST    = df_sam_forward['READID']\n",
    "\n",
    "df_sam_index = df_sam.set_index(['SRRID','READID','PAIRID'])\n",
    "df_sam_index = df_sam_index.sort([3])\n",
    "\n",
    "def cigar_parse(cigar):\n",
    "    match = re.findall(r'(\\d+)(\\w)', cigar)\n",
    "    return match\n",
    "\n",
    "# SVG\n",
    "rect = '<rect x=\"%d\" y=\"%d\" width=\"%d\" height=\"%d\" style=\"fill:%s;stroke:%s;stroke-width:1;fill-opacity:0.8;stroke-opacity:0.3\" />'\n",
    "line = '<line x1=\"%d\" y1=\"%d\" x2=\"%d\" y2=\"%d\" stroke=\"%s\" stroke-width=\"1\" />'\n",
    "text = '<text x=\"%d\"  y=\"%d\" style=\"font-family:Arial;font-size:%d;stroke:black;fill:black;\">%s</text>'\n",
    "\n",
    "Outfile = open('%s.svg'%genename,'w')\n",
    "\n",
    "each_height   = 5\n",
    "each_space    = 2\n",
    "box_height    = 3\n",
    "bridge_height = 2\n",
    "genestart     = 3\n",
    "\n",
    "total_canvas_rows = 100 + genestart# 2 for gene model\n",
    "\n",
    "canvas_width = 1000\n",
    "canvas_height = total_canvas_rows*(each_height+each_space)\n",
    "\n",
    "real_width = right-left+1+100+100\n",
    "\n",
    "print('''<svg height=\"%d\" width=\"%d\">'''%(canvas_height,canvas_width),file=Outfile)\n",
    "gene_space = np.zeros([total_canvas_rows,real_width]) # 100 for paired end stretching out of border\n",
    "\n",
    "############################################################################ drawing space set\n",
    "def get_ratio(x):\n",
    "    return int(float(x)/float(real_width) * canvas_width)\n",
    "\n",
    "\n",
    "df = df_gff_index.loc[(genename,'1')]\n",
    "df = df.reset_index()\n",
    "df = df.sort(3)\n",
    "  \n",
    "fill_CDS   = 'yellow'\n",
    "strock_CDS = 'black'\n",
    "\n",
    "fill_UTR   = 'blue'\n",
    "strock_UTR = 'black'\n",
    "\n",
    "\n",
    "\n",
    "for i in df.index:\n",
    "    if df.loc[i][2] == 'mRNA':\n",
    "        x1 = get_ratio(df.loc[i][3] - left + 100)\n",
    "        y1 = int(each_height/2) + genestart + 1\n",
    "        x2 = get_ratio(df.loc[i][4] - left + 100)\n",
    "        y2 = int(each_height/2) + genestart + 1\n",
    "        print(line%(x1,y1,x2,y2,'black'),file=Outfile)\n",
    "    elif 'UTR' in df.loc[i][2]:\n",
    "        x1 = get_ratio(df.loc[i][3] - left + 100)\n",
    "        y1 = genestart + 1 \n",
    "        w  = get_ratio(df.loc[i][4] - df.loc[i][3])\n",
    "        h  = each_height\n",
    "        print(rect%(x1,y1,w,h,fill_UTR,strock_UTR),file=Outfile)\n",
    "    elif 'CDS' in df.loc[i][2]:\n",
    "        x1 = get_ratio(df.loc[i][3] - left + 100)\n",
    "        y1 = genestart + 1\n",
    "        w  = get_ratio(df.loc[i][4] - df.loc[i][3])\n",
    "        h  = each_height\n",
    "        print(rect%(x1,y1,w,h,fill_CDS,strock_CDS),file=Outfile)\n",
    "\n",
    "print(text%(15,genestart+each_height*4,10,genename),file=Outfile)\n",
    "\n",
    "########################################################################## gene model drawing\n",
    "\n",
    "\n",
    "SRRID_list = set(df_sam_index.index.get_level_values('SRRID'))\n",
    "readnumber = 0\n",
    "bPass = 1\n",
    "for SRRID in SRRID_list:\n",
    "    READID_list = df_sam_index.loc[SRRID].index.get_level_values('READID')\n",
    "    for READID in READID_list:\n",
    "          \n",
    "        df = df_sam_index.loc[(SRRID,READID)]\n",
    "        if len(df) == 2:\n",
    "            pass\n",
    "        else: continue\n",
    "        ##################################### cigar check start\n",
    "        gap_length = 0\n",
    "        for cigar in df[5]:\n",
    "            cigarM       = cigar_parse(cigar)\n",
    "            #print(cigarM)\n",
    "            cigarstrings = [x[1] for x in cigarM]\n",
    "            cigarvalues  = [x[0] for x in cigarM]\n",
    "            for i, cigarstring in enumerate(cigarstrings):\n",
    "                if cigarstring == 'N' or cigarstring == 'I':\n",
    "                    gap_length += int(cigarvalues[i])\n",
    "                    \n",
    "            \n",
    "        fragstart      = min(df[3])-left+100\n",
    "        fragmentsize   = max(df[8])\n",
    "        fragmentsize_r = min(df[8])\n",
    "        insertsize     = fragmentsize - gap_length - 200\n",
    "        \n",
    "        if insertsize < 10:\n",
    "            continue\n",
    "        if fragmentsize_r > 0 :\n",
    "            continue\n",
    "        ##################################################################### Error remove\n",
    "        srow = None\n",
    "        for nrow, row in enumerate(gene_space):\n",
    "            if nrow <= 3:\n",
    "                continue\n",
    "            if max(row[fragstart:fragstart+fragmentsize]) > 0 :\n",
    "                continue\n",
    "            else:\n",
    "                srow = nrow\n",
    "                break\n",
    "        if srow == None:\n",
    "            continue\n",
    "        #print(srow)\n",
    "        ##################################################################### Row selection \n",
    "        gene_space[srow,fragstart:fragstart+fragmentsize] += 1\n",
    "        readnumber = srow\n",
    "        #print(readnumber)\n",
    "        for i in df.index:\n",
    "            fragsize          = df.loc[i][8]\n",
    "            startpoint        = df.loc[i][3]\n",
    "            secondstartpoint  = df.loc[i][7]\n",
    "            cigar        = df.loc[i][5]\n",
    "            cigarM       = cigar_parse(cigar)\n",
    "            #print(cigarM)\n",
    "            cigarstrings = [x[1] for x in cigarM]\n",
    "            cigarvalues  = [x[0] for x in cigarM] \n",
    "            ##################################################################\n",
    "            coverleng = 0\n",
    "            for j, cigarstring in enumerate(cigarstrings):\n",
    "                if cigarstring == 'M' or cigarstring == 'N' or cigarstring == 'I':\n",
    "                    coverleng += int(cigarvalues[j])\n",
    "            if fragsize > 0:\n",
    "                x1 = get_ratio(startpoint - left + 100 + coverleng)\n",
    "                y1 = readnumber * (each_height + each_space) + int(each_height/2)\n",
    "            \n",
    "                y2 = readnumber * (each_height + each_space) + int(each_height/2) + int(each_height/2)\n",
    "                x3 = get_ratio(secondstartpoint - left + 100)\n",
    "                x2 = (x3 + x1)/2\n",
    "                print(line%(x1,y1,x2,y2,'black'),file=Outfile)\n",
    "                print(line%(x2,y2,x3,y1,'black'),file=Outfile)\n",
    "            ################################################################# paired end line\n",
    "            #print(i)\n",
    "            w = 0\n",
    "            #print (cigarstrings)\n",
    "            #print (cigarvalues)\n",
    "            for j, cigarstring in enumerate(cigarstrings):\n",
    "                if cigarstring == 'M':\n",
    "                    x = get_ratio(w + startpoint - left + 100)\n",
    "                    #print(x)\n",
    "                    if x < 0:\n",
    "                        continue\n",
    "                    y = readnumber * (each_height + each_space)\n",
    "                    \n",
    "                    h = int(each_height)\n",
    "                    ew = int(cigarvalues[j]) # each width\n",
    "                    w += ew\n",
    "                    #print(x,y,w,h)\n",
    "                    print(rect%(x,y,get_ratio(ew),h,'pink','black'),file=Outfile)\n",
    "                    \n",
    "                elif cigarstring == 'N' or cigarstring == 'I':\n",
    "                    x1 = get_ratio(w + startpoint - left + 100)\n",
    "                    if x1 < 0:\n",
    "                        continue\n",
    "                    y1 = readnumber * (each_height + each_space) + int(each_height/2)\n",
    "                    ew = int(cigarvalues[j]) # each width\n",
    "                    w += ew\n",
    "                    x2 = get_ratio(w + startpoint - left + 100)\n",
    "                    \n",
    "                    print(line%(x1,y1,x2,y1,'red'),file=Outfile)\n",
    "            \n",
    "        \n",
    "print('''</svg>\n",
    "''',file=Outfile)\n",
    "\n",
    "Outfile.close()               "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "문제 발견! 만약에 insert size로 필터를 하고 싶을때 fragment size로 필터하려고 하면 문제가 생김. 왜냐면 spliced part 까지 포함해서 fragment size를 기록하므로 아무리 큰값으로 필터를 해봐야 여전히 insert size가 readlength 보다 작은 애들이 포함되게 된다. 저놈들이 그림에서 보이는 overlapped reads 웃긴건 저런놈들이 잇는 자리는 뭔가 일관되지 않다는것. 개떡같네 정말 ㅅㅂ. 일단 코드에 insert size를 정확하게 계산하는 부분을 넣도록 하자. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "from IPython.display import SVG, display\n",
    "display(SVG('./%s.svg'%genename))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_sam_index.loc[('SRR2132433','18073440')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
