{
 "cells": [
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 1,
=======
   "execution_count": 2,
>>>>>>> 5e86c6dde92677b7608bb27acc021626231a37d4
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "file_gff_old = './Creinhardtii_281_v5.5.gene.gff3'\n",
=======
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "file_gff_old = '../../../genemodel_correction/Creinhardtii_281_v5.5.gene.gff3'\n",
>>>>>>> 5e86c6dde92677b7608bb27acc021626231a37d4
    "df_gff_old = pd.read_csv(file_gff_old,sep='\\t',header=None,skiprows=2)\n",
    "mask = (df_gff_old[2] == 'gene')\n",
    "df_gff_old_gene = df_gff_old[mask] \n",
    "\n",
<<<<<<< HEAD
    "file_gff_new = '../braker/braker.chlamy20151203.augustus.gff.rmsharp.gff.parsed.gff'\n",
=======
    "file_gff_new = '../../../braker/braker.chlamy20151203.augustus.gff.rmsharp.gff.parsed.gff'\n",
>>>>>>> 5e86c6dde92677b7608bb27acc021626231a37d4
    "df_gff_new = pd.read_csv(file_gff_new,sep='\\t',header=None)\n",
    "mask = (df_gff_new[2] == 'gene')\n",
    "df_gff_new_gene = df_gff_new[mask]\n",
    "\n",
    "df_gff_old[9] = df_gff_old[8].apply(lambda x:x.split(';')[0].replace('ID=',''))\n",
    "df_gff_old = df_gff_old.set_index(9)\n",
    "df_gff_new[9] = df_gff_new[8].apply(lambda x:x.split(';')[0].replace('ID=',''))\n",
    "df_gff_new = df_gff_new.set_index(9)\n",
    "\n",
    "df_gff_new_gene[9] = df_gff_new_gene[3].apply(lambda x:'_'+str(int(x/100000)))\n",
    "df_gff_new_gene[10] = df_gff_new_gene[0] + df_gff_new_gene[9]"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 3,
=======
   "execution_count": 4,
>>>>>>> 5e86c6dde92677b7608bb27acc021626231a37d4
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python2.7/dist-packages/ipykernel/__main__.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# ID=Cre01.g000017.t1.1.v5.5;Name=Cre01.g000017.t1.1;pacid=30789166;longest=1;Parent=Cre01.g000017.v5.5\n",
    "mask = (df_gff_old[2] == 'mRNA')\n",
    "df_gff_old_mRNA = df_gff_old[mask] \n",
    "df_gff_old_mRNA[10] = df_gff_old_mRNA[8].apply(lambda x : x.split(';')[3].replace('longest=',''))\n",
    "\n",
    "mask = (df_gff_old[2] == 'CDS')\n",
    "df_gff_old_CDS = df_gff_old[mask]\n"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 4,
=======
   "execution_count": null,
>>>>>>> 5e86c6dde92677b7608bb27acc021626231a37d4
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## Old gene model parsing\n",
    "dicGN2CDSloc = {}\n",
    "for i in df_gff_old_CDS.index:\n",
    "    genename = '.'.join(i.split('.')[0:6])\n",
    "    chromosome = df_gff_old_CDS.loc[i][0]\n",
    "    left = int(df_gff_old_CDS.loc[i][3])\n",
    "    right = int(df_gff_old_CDS.loc[i][4])\n",
    "    if df_gff_old_mRNA.loc[genename][10] == '1':\n",
    "        pass\n",
    "    else: continue\n",
    "    try:\n",
    "        dicGN2CDSloc[genename][1] += [left,right]\n",
    "    except KeyError:\n",
    "        dicGN2CDSloc[genename] = [chromosome,[left,right]]\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Gene model comparison classification**\n",
    "- exact same\n",
    "- near same\n",
    "- exact split/merge\n",
    "- near split/merge\n",
    "- differ"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "13000\n",
      "14000\n",
      "15000\n",
      "16000\n",
      "17000\n"
     ]
    }
   ],
=======
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
>>>>>>> 5e86c6dde92677b7608bb27acc021626231a37d4
   "source": [
    "g = 0 \n",
    "blocksize = 500\n",
    "dic = {'genename' : [],\n",
    "       'same'     : [],\n",
    "       'nearsame' : [],\n",
    "       'leftspan' : [],\n",
    "       'rightspan': [],\n",
    "       'harbor'   : [],\n",
    "       'leftanchor' : [],\n",
    "       'rightanchor': [],\n",
    "       'matched_genes' : []}\n",
    "for key in dicGN2CDSloc:\n",
    "    genename_old   = '.'.join(key.split('.')[0:2])\n",
    "    g += 1\n",
    "    if g % 1000 == 0:\n",
    "        print g\n",
    "    chromosome_old = dicGN2CDSloc[key][0]\n",
    "    loclist = dicGN2CDSloc[key][1]\n",
    "    loclist.sort()\n",
    "    left_old       = loclist[0]\n",
    "    right_old      = loclist[-1]\n",
    "    mask = (df_gff_new_gene[10] == chromosome_old+'_'+str(int(left_old/100000)))\n",
    "    df_gff_new_gene_chr = df_gff_new_gene[mask]\n",
    "    dic['genename'].append(genename_old)\n",
    "    dic['same'].append(0)\n",
    "    dic['nearsame'].append(0)\n",
    "    dic['leftspan'].append(0)\n",
    "    dic['rightspan'].append(0)\n",
    "    dic['harbor'].append(0)\n",
    "    dic['leftanchor'].append(0)\n",
    "    dic['rightanchor'].append(0) \n",
    "    dic['matched_genes'].append([])\n",
    "    matched_genes = []\n",
    "    for i in df_gff_new_gene_chr.index:\n",
    "        genename_new = df_gff_new_gene_chr.loc[i][8].replace('ID=','')\n",
    "        left_new  = df_gff_new_gene_chr.loc[i][3]\n",
    "        right_new = df_gff_new_gene_chr.loc[i][4]\n",
    "        if left_new == left_old and right_new == right_old:\n",
    "            dic['same'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif abs(left_new - left_old) < blocksize and abs(right_new - right_old) < blocksize:\n",
    "            dic['nearsame'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        \n",
    "        elif left_old == left_new and left_old < right_new < right_old:\n",
    "            dic['leftanchor'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif right_old == right_new and left_old < left_new < right_old:\n",
    "            dic['rightanchor'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "            \n",
    "        elif left_new < left_old < right_new and right_new < right_old:\n",
    "            dic['leftspan'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif left_new < right_old < right_new and left_new > left_old:\n",
    "            dic['rightspan'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif left_old < left_new and right_new < right_old:\n",
    "            dic['harbor'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "df_table = pd.DataFrame(dic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "13000\n",
      "14000\n",
      "15000\n",
      "16000\n",
      "17000\n",
      "18000\n"
     ]
    }
   ],
   "source": [
    "g = 0 \n",
    "blocksize = 500\n",
    "dic = {'genename' : [],\n",
    "       'same'     : [],\n",
    "       'nearsame' : [],\n",
    "       'leftspan' : [],\n",
    "       'rightspan': [],\n",
    "       'harbor'   : [],\n",
    "       'leftanchor' : [],\n",
    "       'rightanchor': [],\n",
    "       'matched_genes' : []}\n",
    "for i in df_gff_new_gene.index:\n",
    "    genename_old   = df_gff_new_gene.loc[i][8].replace('ID=','')\n",
    "    chromosome_old = df_gff_new_gene.loc[i][0]\n",
    "    left_old       = df_gff_new_gene.loc[i][3]\n",
    "    right_old      = df_gff_new_gene.loc[i][4]\n",
    "    g += 1\n",
    "    if g % 1000 == 0:\n",
    "        print g\n",
    "    dic['genename'].append(genename_old)\n",
    "    dic['same'].append(0)\n",
    "    dic['nearsame'].append(0)\n",
    "    dic['leftspan'].append(0)\n",
    "    dic['rightspan'].append(0)\n",
    "    dic['harbor'].append(0)\n",
    "    dic['leftanchor'].append(0)\n",
    "    dic['rightanchor'].append(0) \n",
    "    dic['matched_genes'].append([])\n",
    "    matched_genes = []\n",
    "    for genename_new in dicGN2CDSloc:\n",
    "        chromosome_new = dicGN2CDSloc[genename_new][0]\n",
    "        if chromosome_new != chromosome_old:\n",
    "            continue\n",
    "        loclist   = dicGN2CDSloc[genename_new][1]\n",
    "        loclist.sort()\n",
    "        left_new  = loclist[0]\n",
    "        right_new = loclist[-1]  \n",
    "        if left_new == left_old and right_new == right_old:\n",
    "            dic['same'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif abs(left_new - left_old) < blocksize and abs(right_new - right_old) < blocksize:\n",
    "            dic['nearsame'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        \n",
    "        elif left_old == left_new and left_old < right_new < right_old:\n",
    "            dic['leftanchor'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif right_old == right_new and left_old < left_new < right_old:\n",
    "            dic['rightanchor'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "            \n",
    "        elif left_new < left_old < right_new and right_new < right_old:\n",
    "            dic['leftspan'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif left_new < right_old < right_new and left_new > left_old:\n",
    "            dic['rightspan'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "        elif left_old < left_new and right_new < right_old:\n",
    "            dic['harbor'][-1] += 1\n",
    "            dic['matched_genes'][-1].append(genename_new)\n",
    "df_table_rv = pd.DataFrame(dic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "file_neighbor_new = 'gene_cor_neighbor.ver1.txt'\n",
    "df_neighbor_new = pd.read_csv(file_neighbor_new,sep='\\t',index_col='gene')\n",
    "file_neighbor_old = 'gene_cor_neighbor.txt'\n",
    "df_neighbor_old = pd.read_csv(file_neighbor_old,sep='\\t',index_col='gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneLeft</th>\n",
       "      <th>geneRight</th>\n",
       "      <th>GeneLeftCor</th>\n",
       "      <th>GeneRightCor</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Cre30.g757997</th>\n",
       "      <td>None</td>\n",
       "      <td>Cre30.g758047</td>\n",
       "      <td>None</td>\n",
       "      <td>0.511579398584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre30.g758047</th>\n",
       "      <td>Cre30.g757997</td>\n",
       "      <td>Cre30.g758097</td>\n",
       "      <td>0.511579398584</td>\n",
       "      <td>-0.607835652029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre30.g758097</th>\n",
       "      <td>Cre30.g758047</td>\n",
       "      <td>Cre30.g758147</td>\n",
       "      <td>-0.607835652029</td>\n",
       "      <td>-0.2254877808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre30.g758147</th>\n",
       "      <td>Cre30.g758097</td>\n",
       "      <td>Cre30.g758197</td>\n",
       "      <td>-0.2254877808</td>\n",
       "      <td>-0.152443950461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre30.g758197</th>\n",
       "      <td>Cre30.g758147</td>\n",
       "      <td>Cre30.g758247</td>\n",
       "      <td>-0.152443950461</td>\n",
       "      <td>-0.142834785444</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    geneLeft      geneRight      GeneLeftCor     GeneRightCor\n",
       "gene                                                                         \n",
       "Cre30.g757997           None  Cre30.g758047             None   0.511579398584\n",
       "Cre30.g758047  Cre30.g757997  Cre30.g758097   0.511579398584  -0.607835652029\n",
       "Cre30.g758097  Cre30.g758047  Cre30.g758147  -0.607835652029    -0.2254877808\n",
       "Cre30.g758147  Cre30.g758097  Cre30.g758197    -0.2254877808  -0.152443950461\n",
       "Cre30.g758197  Cre30.g758147  Cre30.g758247  -0.152443950461  -0.142834785444"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neighbor_old.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python2.7/dist-packages/ipykernel/__main__.py:21: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/usr/local/lib/python2.7/dist-packages/ipykernel/__main__.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    }
   ],
   "source": [
    "def get_cor(genelist,df):\n",
    "    try:\n",
    "        genelist = [int(x.replace('g','')) for x in list(genelist)]\n",
    "        genelist.sort()\n",
    "    except ValueError:\n",
    "        genelist = ['.'.join(x.split('.')[0:2]) for x in genelist]\n",
    "    cor = []\n",
    "    for i, geneleft in enumerate(genelist):\n",
    "        try:\n",
    "            generight = genelist[i+1]\n",
    "        except IndexError:\n",
    "            continue\n",
    "        try:\n",
    "            cor.append(df.loc[geneleft]['GeneRightCor'])\n",
    "        except KeyError:\n",
    "            continue\n",
    "    return cor\n",
    "\n",
    "mask = (df_table['leftanchor'] > 0) & (df_table['rightanchor'] > 0)\n",
    "df_table_split = df_table[mask]\n",
    "df_table_split['cor_list'] = df_table_split['matched_genes'].apply(get_cor,args=(df_neighbor_new,))\n",
    "\n",
    "mask = (df_table_rv['leftanchor'] > 0) & (df_table_rv['rightanchor'] > 0)\n",
    "df_table_rv_split = df_table_rv[mask]\n",
    "df_table_rv_split['cor_list'] = df_table_rv_split['matched_genes'].apply(get_cor,args=(df_neighbor_old,))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "filename_mayra_plate_index = 'mayra.plate.index.txt'\n",
    "df_mayra_plate_index = pd.read_csv(filename_mayra_plate_index,sep='\\t')\n",
    "\n",
    "file_mayra_pcrlist = 'mayra.pcr.list.txt'\n",
    "df_mayra_pcrlist = pd.read_csv(file_mayra_pcrlist,sep='\\t')\n",
    "\n",
    "df_mayra_plate_index_platename = df_mayra_plate_index.set_index(['platename','Pos.Nr'])\n",
    "\n",
    "def get_genename(df):\n",
    "    x = df['platename']\n",
    "    y = df['Pos.Nr']\n",
    "    genename = df_mayra_plate_index_platename.loc[(x,y)]['ChlreGene_ID55']\n",
    "    return genename\n",
    "df_mayra_pcrlist['genename'] = df_mayra_pcrlist.apply(get_genename,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "filename_cds_cov = 'est_cov.ver2.txt'\n",
    "df_cds_cov = pd.read_csv(filename_cds_cov,sep='\\t',header=None)\n",
    "df_cds_cov['genename'] = df_cds_cov[0].apply(lambda x : '.'.join(x.split('.')[0:2]))\n",
    "df_cds_cov_indexed = df_cds_cov.set_index('genename')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    Cre03.g146147.t1.1.v5.5\n",
       "1                          0\n",
       "2                        0.0\n",
       "3                        0.0\n",
       "4                        0.0\n",
       "5              Cre03.g146127\n",
       "6              Cre03.g146167\n",
       "7            -0.405723143605\n",
       "8            -0.530418970506\n",
       "Name: Cre03.g146147, dtype: object"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cds_cov_indexed.loc['Cre03.g146147']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>platename</th>\n",
       "      <th>Pos.Nr</th>\n",
       "      <th>Position</th>\n",
       "      <th>ORF_Size5</th>\n",
       "      <th>right size</th>\n",
       "      <th>band - wrong size</th>\n",
       "      <th>no band</th>\n",
       "      <th>nur right band</th>\n",
       "      <th>nur mittlere band</th>\n",
       "      <th>no band.1</th>\n",
       "      <th>sum</th>\n",
       "      <th>genename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P1kCre1+2</td>\n",
       "      <td>1</td>\n",
       "      <td>A1</td>\n",
       "      <td>579</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Cre11.g468300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P1kCre1+2</td>\n",
       "      <td>2</td>\n",
       "      <td>B1</td>\n",
       "      <td>369</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Cre06.g260400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P1kCre1+2</td>\n",
       "      <td>3</td>\n",
       "      <td>C1</td>\n",
       "      <td>393</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Cre01.g025025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P1kCre1+2</td>\n",
       "      <td>4</td>\n",
       "      <td>D1</td>\n",
       "      <td>417</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Cre02.g077500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P1kCre1+2</td>\n",
       "      <td>5</td>\n",
       "      <td>E1</td>\n",
       "      <td>435</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Cre07.g330950</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   platename  Pos.Nr Position  ORF_Size5  right size  band - wrong size  \\\n",
       "0  P1kCre1+2       1       A1        579           1                NaN   \n",
       "1  P1kCre1+2       2       B1        369           1                NaN   \n",
       "2  P1kCre1+2       3       C1        393           0                NaN   \n",
       "3  P1kCre1+2       4       D1        417           1                NaN   \n",
       "4  P1kCre1+2       5       E1        435           0                NaN   \n",
       "\n",
       "   no band  nur right band  nur mittlere band  no band.1  sum       genename  \n",
       "0      NaN               1                  0          0    1  Cre11.g468300  \n",
       "1      NaN               1                  0          0    1  Cre06.g260400  \n",
       "2        1               0                  0          1    1  Cre01.g025025  \n",
       "3      NaN               1                  0          0    1  Cre02.g077500  \n",
       "4        1               0                  0          1    1  Cre07.g330950  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mayra_pcrlist.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_mayra_pcrlist_index = df_mayra_pcrlist.set_index('genename')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python2.7/dist-packages/ipykernel/__main__.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    }
   ],
   "source": [
    "def get_pcr(x):\n",
    "    try:\n",
    "        return df_mayra_pcrlist_index.loc[x]['right size']\n",
    "    except KeyError:\n",
    "        return None\n",
    "df_table_split['mayra_PCR'] = df_table['genename'].apply(get_pcr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    13\n",
       "0     9\n",
       "Name: mayra_PCR, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_table_split['mayra_PCR'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_table_split_index = df_table_split.set_index('genename')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_table_split.to_csv('cre.splitlist.txt',sep='\\t')\n",
    "\n",
    "df_table_rv_split.to_csv('cre.mergelist.txt',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "genes = []\n",
    "for each in df_table_rv_split['matched_genes']:\n",
    "    genes += each\n",
    "dic = {'genename_pre':genes}\n",
    "df_merged_list = pd.DataFrame(dic)\n",
    "df_merged_list['genename'] = df_merged_list['genename_pre'].apply(lambda x : '.'.join(x.split('.')[0:2]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_pcr(x):\n",
    "    try:\n",
    "        return df_mayra_pcrlist_index.loc[x]['right size']\n",
    "    except KeyError:\n",
    "        return None\n",
    "df_merged_list['pcr_result'] = df_merged_list['genename'].apply(get_pcr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    59\n",
       "0    25\n",
       "Name: pcr_result, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merged_list['pcr_result'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_merged_list_index = df_merged_list.set_index('genename')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_cdscov(x):\n",
    "    try:\n",
    "        a = float(df_cds_cov_indexed.loc[x][1])\n",
    "        return df_cds_cov_indexed.loc[x][1]\n",
    "    except :\n",
    "            return df_cds_cov_indexed.loc[x][1][0]\n",
    "\n",
    "df_table['CDScov'] = df_table['genename'].apply(get_cdscov) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "filename = 'cre.initialgeneset8637.txt'\n",
    "df_initialgeneset = pd.read_csv(filename,header=None)\n",
    "genelist_initialgeneset = list(df_initialgeneset[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_initial(x):\n",
    "    if x in genelist_initialgeneset:\n",
    "        return 1\n",
    "    else: return 0\n",
    "df_table['initial_geneset'] = df_table['genename'].apply(get_initial)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "genelist_mayra_plate = list(df_mayra_plate_index['ChlreGene_ID55'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_plate(x):\n",
    "    if x in genelist_mayra_plate:\n",
    "        return 1\n",
    "    else: return 0 \n",
    "df_table['mayra_plate'] = df_table['genename'].apply(get_plate)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_pcr(x):\n",
    "    try:\n",
    "        return df_mayra_pcrlist_index.loc[x]['right size']\n",
    "    except KeyError:\n",
    "        return None\n",
    "df_table['mayra_pcr'] = df_table['genename'].apply(get_pcr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_merge(x):\n",
    "    try:\n",
    "        a = df_merged_list_index.loc[x]\n",
    "        return 1\n",
    "    except KeyError:\n",
    "        return 0\n",
    "def get_split(x):\n",
    "    try:\n",
    "        a = df_table_split_index.loc[x]\n",
    "        return 1\n",
    "    except KeyError:\n",
    "        return 0\n",
    "df_table['merged'] = df_table['genename'].apply(get_merge)\n",
    "df_table['split'] = df_table['genename'].apply(get_split)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_table['start'] = df_table['genename'].apply(lambda x : df_gff_old.loc[x+'.v5.5'][3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Cre01.g000017.v5.5</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>gene</td>\n",
       "      <td>18766</td>\n",
       "      <td>20237</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=Cre01.g000017.v5.5;Name=Cre01.g000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre01.g000017.t1.1.v5.5</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>18766</td>\n",
       "      <td>20237</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=Cre01.g000017.t1.1.v5.5;Name=Cre01.g000017....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre01.g000017.t1.1.v5.5.five_prime_UTR.1</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>18766</td>\n",
       "      <td>19162</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=Cre01.g000017.t1.1.v5.5.five_prime_UTR.1;Pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre01.g000017.t1.1.v5.5.CDS.1</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>CDS</td>\n",
       "      <td>19163</td>\n",
       "      <td>19178</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=Cre01.g000017.t1.1.v5.5.CDS.1;Parent=Cre01....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cre01.g000017.t1.1.v5.5.CDS.2</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>phytozomev10</td>\n",
       "      <td>CDS</td>\n",
       "      <td>19329</td>\n",
       "      <td>19948</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=Cre01.g000017.t1.1.v5.5.CDS.2;Parent=Cre01....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     0             1  \\\n",
       "9                                                                      \n",
       "Cre01.g000017.v5.5                        chromosome_1  phytozomev10   \n",
       "Cre01.g000017.t1.1.v5.5                   chromosome_1  phytozomev10   \n",
       "Cre01.g000017.t1.1.v5.5.five_prime_UTR.1  chromosome_1  phytozomev10   \n",
       "Cre01.g000017.t1.1.v5.5.CDS.1             chromosome_1  phytozomev10   \n",
       "Cre01.g000017.t1.1.v5.5.CDS.2             chromosome_1  phytozomev10   \n",
       "\n",
       "                                                       2      3      4  5  6  \\\n",
       "9                                                                              \n",
       "Cre01.g000017.v5.5                                  gene  18766  20237  .  +   \n",
       "Cre01.g000017.t1.1.v5.5                             mRNA  18766  20237  .  +   \n",
       "Cre01.g000017.t1.1.v5.5.five_prime_UTR.1  five_prime_UTR  18766  19162  .  +   \n",
       "Cre01.g000017.t1.1.v5.5.CDS.1                        CDS  19163  19178  .  +   \n",
       "Cre01.g000017.t1.1.v5.5.CDS.2                        CDS  19329  19948  .  +   \n",
       "\n",
       "                                          7  \\\n",
       "9                                             \n",
       "Cre01.g000017.v5.5                        .   \n",
       "Cre01.g000017.t1.1.v5.5                   .   \n",
       "Cre01.g000017.t1.1.v5.5.five_prime_UTR.1  .   \n",
       "Cre01.g000017.t1.1.v5.5.CDS.1             0   \n",
       "Cre01.g000017.t1.1.v5.5.CDS.2             2   \n",
       "\n",
       "                                                                                          8  \n",
       "9                                                                                            \n",
       "Cre01.g000017.v5.5                                 ID=Cre01.g000017.v5.5;Name=Cre01.g000017  \n",
       "Cre01.g000017.t1.1.v5.5                   ID=Cre01.g000017.t1.1.v5.5;Name=Cre01.g000017....  \n",
       "Cre01.g000017.t1.1.v5.5.five_prime_UTR.1  ID=Cre01.g000017.t1.1.v5.5.five_prime_UTR.1;Pa...  \n",
       "Cre01.g000017.t1.1.v5.5.CDS.1             ID=Cre01.g000017.t1.1.v5.5.CDS.1;Parent=Cre01....  \n",
       "Cre01.g000017.t1.1.v5.5.CDS.2             ID=Cre01.g000017.t1.1.v5.5.CDS.2;Parent=Cre01....  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gff_old.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_break(x):\n",
    "    results= []\n",
    "    if len(x) < 2:\n",
    "        return None\n",
    "    else:\n",
    "        for g in x:\n",
    "            if g == x[0]:\n",
    "                results.append(df_gff_new.loc[g][4])\n",
    "            elif g== x[-1]:\n",
    "                results.append(df_gff_new.loc[g][3])\n",
    "            else:\n",
    "                results.append(df_gff_new.loc[g][3])\n",
    "                results.append(df_gff_new.loc[g][4])\n",
    "    return results\n",
    "                        \n",
    "        \n",
    "df_table['break'] = df_table['matched_genes'].apply(get_break)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_table.to_csv('cre.genemodelcomp.total.txt',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_table.to_pickle('cre.genemodelcomp.total.pk')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genename</th>\n",
       "      <th>harbor</th>\n",
       "      <th>leftanchor</th>\n",
       "      <th>leftspan</th>\n",
       "      <th>matched_genes</th>\n",
       "      <th>nearsame</th>\n",
       "      <th>rightanchor</th>\n",
       "      <th>rightspan</th>\n",
       "      <th>same</th>\n",
       "      <th>CDScov</th>\n",
       "      <th>initial_geneset</th>\n",
       "      <th>mayra_plate</th>\n",
       "      <th>mayra_pcr</th>\n",
       "      <th>merged</th>\n",
       "      <th>split</th>\n",
       "      <th>break</th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cre15.g636750</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g6719]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>495422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Cre13.g575333</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g5477]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1850658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cre02.g076250</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g1373]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>440815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cre09.g400664</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g3726]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>5552067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Cre03.g146147</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g16362]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>736244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Cre06.g253753</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>710740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Cre06.g278118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g15274]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>3564037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Cre08.g366800</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g8270]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.988551</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1856824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Cre07.g337516</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g7536]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.980938</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>3679849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Cre11.g467547</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g11396]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>167646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Cre09.g410150</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>6886095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Cre06.g299700</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g15961]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>7413232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Cre13.g570751</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g5385]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1272317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Cre07.g344950</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g7703]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4701800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Cre06.g302551</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g16024]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.844544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>7802226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Cre07.g315850</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g6976]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>490371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Cre12.g531950</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g13009]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>5596543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Cre15.g638700</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1029830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Cre02.g078700</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g1437]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.999057</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>759828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Cre06.g278206</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g15361]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4031066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Cre51.g761547</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g14076]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.905318</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Cre08.g383702</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g8725]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4668265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Cre14.g624800</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g6407]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>2436789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Cre03.g173700</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g16954]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4161665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Cre05.g240650</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g14547]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>3336234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Cre06.g270150</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g15073]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>2670238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Cre02.g095052</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>3072920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Cre04.g216850</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g17898]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>144030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Cre12.g547700</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g13522]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>8524114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Cre12.g541300</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g13671]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.998088</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>9266696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17711</th>\n",
       "      <td>Cre12.g557252</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>7345258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17712</th>\n",
       "      <td>Cre01.g009500</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g237]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1763026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17713</th>\n",
       "      <td>Cre01.g031951</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g727]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.993180</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4642355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17714</th>\n",
       "      <td>Cre06.g303350</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g16044]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>7915821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17715</th>\n",
       "      <td>Cre16.g671750</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>6637711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17716</th>\n",
       "      <td>Cre11.g467685</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g11533]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1152260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17717</th>\n",
       "      <td>Cre03.g145887</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g16350]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>671066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17718</th>\n",
       "      <td>Cre17.g713100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g10513]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.059942</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>2184840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17719</th>\n",
       "      <td>Cre01.g044350</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g999]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>6200724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17720</th>\n",
       "      <td>Cre16.g667850</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g9538]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4307717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17721</th>\n",
       "      <td>Cre01.g036500</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g821]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>5237847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17722</th>\n",
       "      <td>Cre12.g502179</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g12447]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>2866626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17723</th>\n",
       "      <td>Cre07.g327800</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g7310]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.105291</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>2292076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17724</th>\n",
       "      <td>Cre02.g095054</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>[g1864]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>3084120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17725</th>\n",
       "      <td>Cre09.g416000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g4045]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>7726054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17726</th>\n",
       "      <td>Cre03.g191300</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g17336]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>6048076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17727</th>\n",
       "      <td>Cre16.g687850</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g9586]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4593327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17728</th>\n",
       "      <td>Cre07.g327650</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g7306]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>2262876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17729</th>\n",
       "      <td>Cre09.g402750</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g2964]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>697454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17730</th>\n",
       "      <td>Cre11.g467559</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g11407]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>272703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17731</th>\n",
       "      <td>Cre09.g399067</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g3653]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>5154883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17732</th>\n",
       "      <td>Cre16.g688638</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g10095]</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>7553486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17733</th>\n",
       "      <td>Cre14.g613050</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[g6116, g6117, g6118]</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>[792845, 793121, 795854, 795863]</td>\n",
       "      <td>788732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17734</th>\n",
       "      <td>Cre06.g286000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g15673]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>5636290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17735</th>\n",
       "      <td>Cre03.g171000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g16899]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>3863048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17736</th>\n",
       "      <td>Cre01.g010400</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g257]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1945728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17737</th>\n",
       "      <td>Cre12.g512400</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g12224]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.996433</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>1757519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17738</th>\n",
       "      <td>Cre03.g178400</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g17071]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>4784212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17739</th>\n",
       "      <td>Cre16.g694402</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[g8824]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>257962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17740</th>\n",
       "      <td>Cre03.g188850</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[g17287]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>5791770</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17741 rows  17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            genename  harbor  leftanchor  leftspan          matched_genes  \\\n",
       "0      Cre15.g636750       0           0         0                [g6719]   \n",
       "1      Cre13.g575333       0           0         0                [g5477]   \n",
       "2      Cre02.g076250       0           0         0                [g1373]   \n",
       "3      Cre09.g400664       0           0         0                [g3726]   \n",
       "4      Cre03.g146147       0           0         0               [g16362]   \n",
       "5      Cre06.g253753       0           0         0                     []   \n",
       "6      Cre06.g278118       0           0         0               [g15274]   \n",
       "7      Cre08.g366800       0           0         0                [g8270]   \n",
       "8      Cre07.g337516       0           0         0                [g7536]   \n",
       "9      Cre11.g467547       0           0         0               [g11396]   \n",
       "10     Cre09.g410150       0           0         0                     []   \n",
       "11     Cre06.g299700       0           0         0               [g15961]   \n",
       "12     Cre13.g570751       0           0         0                [g5385]   \n",
       "13     Cre07.g344950       0           0         0                [g7703]   \n",
       "14     Cre06.g302551       0           0         0               [g16024]   \n",
       "15     Cre07.g315850       0           0         0                [g6976]   \n",
       "16     Cre12.g531950       0           0         0               [g13009]   \n",
       "17     Cre15.g638700       0           0         0                     []   \n",
       "18     Cre02.g078700       0           0         0                [g1437]   \n",
       "19     Cre06.g278206       0           0         0               [g15361]   \n",
       "20     Cre51.g761547       0           0         0               [g14076]   \n",
       "21     Cre08.g383702       0           0         0                [g8725]   \n",
       "22     Cre14.g624800       0           0         0                [g6407]   \n",
       "23     Cre03.g173700       0           0         0               [g16954]   \n",
       "24     Cre05.g240650       0           0         0               [g14547]   \n",
       "25     Cre06.g270150       0           0         0               [g15073]   \n",
       "26     Cre02.g095052       0           0         0                     []   \n",
       "27     Cre04.g216850       0           0         0               [g17898]   \n",
       "28     Cre12.g547700       0           0         0               [g13522]   \n",
       "29     Cre12.g541300       0           0         0               [g13671]   \n",
       "...              ...     ...         ...       ...                    ...   \n",
       "17711  Cre12.g557252       0           0         0                     []   \n",
       "17712  Cre01.g009500       0           0         0                 [g237]   \n",
       "17713  Cre01.g031951       0           0         0                 [g727]   \n",
       "17714  Cre06.g303350       0           0         0               [g16044]   \n",
       "17715  Cre16.g671750       0           0         0                     []   \n",
       "17716  Cre11.g467685       0           0         0               [g11533]   \n",
       "17717  Cre03.g145887       0           0         0               [g16350]   \n",
       "17718  Cre17.g713100       0           0         0               [g10513]   \n",
       "17719  Cre01.g044350       0           0         0                 [g999]   \n",
       "17720  Cre16.g667850       0           0         0                [g9538]   \n",
       "17721  Cre01.g036500       0           0         0                 [g821]   \n",
       "17722  Cre12.g502179       0           0         0               [g12447]   \n",
       "17723  Cre07.g327800       0           0         0                [g7310]   \n",
       "17724  Cre02.g095054       0           0         1                [g1864]   \n",
       "17725  Cre09.g416000       0           0         0                [g4045]   \n",
       "17726  Cre03.g191300       0           0         0               [g17336]   \n",
       "17727  Cre16.g687850       0           0         0                [g9586]   \n",
       "17728  Cre07.g327650       0           0         0                [g7306]   \n",
       "17729  Cre09.g402750       0           0         0                [g2964]   \n",
       "17730  Cre11.g467559       0           0         0               [g11407]   \n",
       "17731  Cre09.g399067       0           0         0                [g3653]   \n",
       "17732  Cre16.g688638       0           0         0               [g10095]   \n",
       "17733  Cre14.g613050       1           1         0  [g6116, g6117, g6118]   \n",
       "17734  Cre06.g286000       0           0         0               [g15673]   \n",
       "17735  Cre03.g171000       0           0         0               [g16899]   \n",
       "17736  Cre01.g010400       0           0         0                 [g257]   \n",
       "17737  Cre12.g512400       0           0         0               [g12224]   \n",
       "17738  Cre03.g178400       0           0         0               [g17071]   \n",
       "17739  Cre16.g694402       0           0         0                [g8824]   \n",
       "17740  Cre03.g188850       0           1         0               [g17287]   \n",
       "\n",
       "       nearsame  rightanchor  rightspan  same    CDScov  initial_geneset  \\\n",
       "0             0            0          0     1  1.000000                1   \n",
       "1             1            0          0     0  1.000000                1   \n",
       "2             0            0          0     1  1.000000                1   \n",
       "3             0            0          0     1  1.000000                0   \n",
       "4             0            0          0     1  0.000000                1   \n",
       "5             0            0          0     0  1.000000                0   \n",
       "6             1            0          0     0  1.000000                0   \n",
       "7             1            0          0     0  0.988551                0   \n",
       "8             0            0          0     1  0.980938                1   \n",
       "9             0            0          0     1  1.000000                1   \n",
       "10            0            0          0     0  1.000000                0   \n",
       "11            0            0          0     1  1.000000                1   \n",
       "12            1            0          0     0  1.000000                0   \n",
       "13            0            0          0     1  1.000000                1   \n",
       "14            1            0          0     0  0.844544                0   \n",
       "15            0            0          0     1  1.000000                1   \n",
       "16            0            0          0     1  1.000000                1   \n",
       "17            0            0          0     0  1.000000                0   \n",
       "18            0            0          0     1  0.999057                1   \n",
       "19            0            0          0     1  1.000000                0   \n",
       "20            1            0          0     0  0.905318                1   \n",
       "21            0            0          0     1  1.000000                1   \n",
       "22            0            0          0     1  0.000000                0   \n",
       "23            0            0          0     1  1.000000                1   \n",
       "24            0            0          0     1  1.000000                1   \n",
       "25            0            0          0     1  1.000000                1   \n",
       "26            0            0          0     0  1.000000                0   \n",
       "27            0            0          0     1  1.000000                1   \n",
       "28            0            0          0     1  1.000000                1   \n",
       "29            0            0          0     1  0.998088                1   \n",
       "...         ...          ...        ...   ...       ...              ...   \n",
       "17711         0            0          0     0  1.000000                1   \n",
       "17712         0            0          0     1  1.000000                1   \n",
       "17713         0            0          0     1  0.993180                1   \n",
       "17714         0            0          0     1  1.000000                0   \n",
       "17715         0            0          0     0  1.000000                1   \n",
       "17716         0            0          0     1  1.000000                1   \n",
       "17717         0            0          0     1  0.777778                1   \n",
       "17718         0            0          0     1  0.059942                0   \n",
       "17719         1            0          0     0  1.000000                0   \n",
       "17720         1            0          0     0  1.000000                1   \n",
       "17721         0            0          0     1  1.000000                0   \n",
       "17722         0            0          0     1  1.000000                0   \n",
       "17723         0            0          0     1  0.105291                1   \n",
       "17724         0            0          0     0  1.000000                0   \n",
       "17725         0            0          0     1  1.000000                0   \n",
       "17726         1            0          0     0  1.000000                0   \n",
       "17727         1            0          0     0  1.000000                1   \n",
       "17728         1            0          0     0  1.000000                1   \n",
       "17729         0            0          0     1  1.000000                1   \n",
       "17730         1            0          0     0  1.000000                0   \n",
       "17731         1            0          0     0  1.000000                1   \n",
       "17732         0            1          0     0  1.000000                1   \n",
       "17733         0            1          0     0  0.833333                0   \n",
       "17734         0            0          0     1  1.000000                0   \n",
       "17735         0            0          0     1  1.000000                0   \n",
       "17736         0            0          0     1  1.000000                0   \n",
       "17737         0            0          0     1  0.996433                1   \n",
       "17738         0            0          0     1  1.000000                1   \n",
       "17739         1            0          0     0  1.000000                0   \n",
       "17740         0            0          0     0  0.000000                1   \n",
       "\n",
       "       mayra_plate  mayra_pcr  merged  split  \\\n",
       "0                0        NaN       0      0   \n",
       "1                0        NaN       0      0   \n",
       "2                0        NaN       0      0   \n",
       "3                0        NaN       0      0   \n",
       "4                1          0       0      0   \n",
       "5                0        NaN       0      0   \n",
       "6                0        NaN       0      0   \n",
       "7                0        NaN       0      0   \n",
       "8                1        NaN       0      0   \n",
       "9                0        NaN       0      0   \n",
       "10               0        NaN       0      0   \n",
       "11               1          1       0      0   \n",
       "12               0        NaN       0      0   \n",
       "13               1          1       0      0   \n",
       "14               0        NaN       0      0   \n",
       "15               0        NaN       0      0   \n",
       "16               0        NaN       0      0   \n",
       "17               0        NaN       1      0   \n",
       "18               0        NaN       0      0   \n",
       "19               0        NaN       0      0   \n",
       "20               0        NaN       0      0   \n",
       "21               0        NaN       0      0   \n",
       "22               0        NaN       0      0   \n",
       "23               1          1       0      0   \n",
       "24               1        NaN       0      0   \n",
       "25               1          1       0      0   \n",
       "26               0        NaN       0      0   \n",
       "27               1          1       0      0   \n",
       "28               0        NaN       0      0   \n",
       "29               0        NaN       0      0   \n",
       "...            ...        ...     ...    ...   \n",
       "17711            1          1       0      0   \n",
       "17712            0        NaN       0      0   \n",
       "17713            0        NaN       0      0   \n",
       "17714            0        NaN       0      0   \n",
       "17715            1          1       1      0   \n",
       "17716            0        NaN       0      0   \n",
       "17717            1          1       0      0   \n",
       "17718            0        NaN       0      0   \n",
       "17719            0        NaN       0      0   \n",
       "17720            1          1       0      0   \n",
       "17721            0        NaN       0      0   \n",
       "17722            0        NaN       0      0   \n",
       "17723            1          0       0      0   \n",
       "17724            0        NaN       0      0   \n",
       "17725            0        NaN       0      0   \n",
       "17726            0        NaN       0      0   \n",
       "17727            1          1       0      0   \n",
       "17728            0        NaN       0      0   \n",
       "17729            0        NaN       0      0   \n",
       "17730            0        NaN       0      0   \n",
       "17731            0        NaN       0      0   \n",
       "17732            0        NaN       0      0   \n",
       "17733            0        NaN       0      1   \n",
       "17734            0        NaN       0      0   \n",
       "17735            0        NaN       0      0   \n",
       "17736            0        NaN       0      0   \n",
       "17737            0        NaN       0      0   \n",
       "17738            0        NaN       0      0   \n",
       "17739            0        NaN       0      0   \n",
       "17740            1        NaN       0      0   \n",
       "\n",
       "                                  break    start  \n",
       "0                                  None   495422  \n",
       "1                                  None  1850658  \n",
       "2                                  None   440815  \n",
       "3                                  None  5552067  \n",
       "4                                  None   736244  \n",
       "5                                  None   710740  \n",
       "6                                  None  3564037  \n",
       "7                                  None  1856824  \n",
       "8                                  None  3679849  \n",
       "9                                  None   167646  \n",
       "10                                 None  6886095  \n",
       "11                                 None  7413232  \n",
       "12                                 None  1272317  \n",
       "13                                 None  4701800  \n",
       "14                                 None  7802226  \n",
       "15                                 None   490371  \n",
       "16                                 None  5596543  \n",
       "17                                 None  1029830  \n",
       "18                                 None   759828  \n",
       "19                                 None  4031066  \n",
       "20                                 None      149  \n",
       "21                                 None  4668265  \n",
       "22                                 None  2436789  \n",
       "23                                 None  4161665  \n",
       "24                                 None  3336234  \n",
       "25                                 None  2670238  \n",
       "26                                 None  3072920  \n",
       "27                                 None   144030  \n",
       "28                                 None  8524114  \n",
       "29                                 None  9266696  \n",
       "...                                 ...      ...  \n",
       "17711                              None  7345258  \n",
       "17712                              None  1763026  \n",
       "17713                              None  4642355  \n",
       "17714                              None  7915821  \n",
       "17715                              None  6637711  \n",
       "17716                              None  1152260  \n",
       "17717                              None   671066  \n",
       "17718                              None  2184840  \n",
       "17719                              None  6200724  \n",
       "17720                              None  4307717  \n",
       "17721                              None  5237847  \n",
       "17722                              None  2866626  \n",
       "17723                              None  2292076  \n",
       "17724                              None  3084120  \n",
       "17725                              None  7726054  \n",
       "17726                              None  6048076  \n",
       "17727                              None  4593327  \n",
       "17728                              None  2262876  \n",
       "17729                              None   697454  \n",
       "17730                              None   272703  \n",
       "17731                              None  5154883  \n",
       "17732                              None  7553486  \n",
       "17733  [792845, 793121, 795854, 795863]   788732  \n",
       "17734                              None  5636290  \n",
       "17735                              None  3863048  \n",
       "17736                              None  1945728  \n",
       "17737                              None  1757519  \n",
       "17738                              None  4784212  \n",
       "17739                              None   257962  \n",
       "17740                              None  5791770  \n",
       "\n",
       "[17741 rows x 17 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "- Same gene list, Near same gene list, Split, Merge  0  1   \n",
    "-   ,  ,      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 387,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "gene_same_list_genename = ['.'.join(x.split('.')[0:2]) for x in gene_same_list]\n",
    "gene_nearsame_genename = ['.'.join(x.split('.')[0:2]) for x in gene_nearsame_list]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "filename_split = 'cre.genemodel.splited.txt'\n",
    "filename_merge = 'cre.genemodel.merged.txt'\n",
    "df_split = pd.read_csv(filename_split,sep='\\t')\n",
    "df_merge = pd.read_csv(filename_merge,sep='\\t')\n",
    "\n",
    "df_split['genename_p'] = df_split['genename'].apply(lambda x : '.'.join(x.split('.')[0:2]))\n",
    "\n",
    "gene_splited_genename = list(df_split['genename_p'])\n",
    "\n",
    "merged_list = []\n",
    "for each in df_result_old['splited']:\n",
    "    merged_list += list(each)\n",
    "\n",
    "gene_merged_genename = ['.'.join(x.split('.')[0:2]) for x in  merged_list]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "Now let's compare with ~8900 genes that are the initial geneset from klaus lab. \n",
    "```python\n",
    "filename = 'cre.initialgeneset8637.txt'\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_deter(x,clist):\n",
    "    if x in clist:\n",
    "        return 1\n",
    "    else: return 0\n",
    "df_initialgeneset['samegenemodel'] = df_initialgeneset[0].apply(get_deter,args=(gene_same_list_genename,)) \n",
    "df_initialgeneset['nearsamegenemodel'] = df_initialgeneset[0].apply(get_deter,args=(gene_nearsame_genename,)) \n",
    "df_initialgeneset['split'] = df_initialgeneset[0].apply(get_deter,args=(gene_splited_genename,)) \n",
    "df_initialgeneset['merge'] = df_initialgeneset[0].apply(get_deter,args=(gene_merged_genename,)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1    5687\n",
      "0    2950\n",
      "Name: samegenemodel, dtype: int64\n",
      "0    6862\n",
      "1    1775\n",
      "Name: nearsamegenemodel, dtype: int64\n",
      "0    8400\n",
      "1     237\n",
      "Name: split, dtype: int64\n",
      "0    8230\n",
      "1     407\n",
      "Name: merge, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print df_initialgeneset['samegenemodel'].value_counts()\n",
    "print df_initialgeneset['nearsamegenemodel'].value_counts()\n",
    "print df_initialgeneset['split'].value_counts()\n",
    "print df_initialgeneset['merge'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Which means that 5697 genes are safe to go withi 8637 genes in initial genelist"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "Now let's compare with 3935 mayra's plate\n",
    "```python\n",
    "filename = 'mayra.plate.index.txt'\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_mayra_plate_index['samegenemodel'] = df_mayra_plate_index['ChlreGene_ID55'].apply(get_deter,args=(gene_same_list_genename,))\n",
    "df_mayra_plate_index['nearsamegenemodel'] = df_mayra_plate_index['ChlreGene_ID55'].apply(get_deter,args=(gene_nearsame_genename,))\n",
    "df_mayra_plate_index['split'] = df_mayra_plate_index['ChlreGene_ID55'].apply(get_deter,args=(gene_splited_genename,))\n",
    "df_mayra_plate_index['merge'] = df_mayra_plate_index['ChlreGene_ID55'].apply(get_deter,args=(gene_merged_genename,))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1    2657\n",
      "0    1274\n",
      "Name: samegenemodel, dtype: int64\n",
      "0    3104\n",
      "1     827\n",
      "Name: nearsamegenemodel, dtype: int64\n",
      "0    3928\n",
      "1       3\n",
      "Name: split, dtype: int64\n",
      "0    3916\n",
      "1      15\n",
      "Name: merge, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print df_mayra_plate_index['samegenemodel'].value_counts()\n",
    "print df_mayra_plate_index['nearsamegenemodel'].value_counts()\n",
    "print df_mayra_plate_index['split'].value_counts()\n",
    "print df_mayra_plate_index['merge'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "which means that 2657 genes in mayra's set would be safe "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "Let's compare with mayra's pcr\n",
    "```python\n",
    "filename = 'mayra.pcr.list.txt'\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_mayra_pcrlist['samegenemodel'] = df_mayra_pcrlist['genename'].apply(get_deter,args=(gene_same_list_genename,))\n",
    "df_mayra_pcrlist['nearsamegenemodel'] = df_mayra_pcrlist['genename'].apply(get_deter,args=(gene_nearsame_genename,))\n",
    "df_mayra_pcrlist['split'] = df_mayra_pcrlist['genename'].apply(get_deter,args=(gene_splited_genename,))\n",
    "df_mayra_pcrlist['merge'] = df_mayra_pcrlist['genename'].apply(get_deter,args=(gene_merged_genename,))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1    2371\n",
      "0    1158\n",
      "Name: samegenemodel, dtype: int64\n",
      "0    2778\n",
      "1     751\n",
      "Name: nearsamegenemodel, dtype: int64\n",
      "0    3472\n",
      "1      57\n",
      "Name: split, dtype: int64\n",
      "0    3417\n",
      "1     112\n",
      "Name: merge, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print df_mayra_pcrlist['samegenemodel'].value_counts()\n",
    "print df_mayra_pcrlist['nearsamegenemodel'].value_counts()\n",
    "print df_mayra_pcrlist['split'].value_counts()\n",
    "print df_mayra_pcrlist['merge'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_mayra_pcrlist.to_csv(file_mayra_pcrlist + '.genemodelcomp.txt',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mask_pcr_P = (df_mayra_pcrlist['right size'] == 1)\n",
    "mask_pcr_F = (df_mayra_pcrlist['right size'] == 0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    1813\n",
       "0     791\n",
       "Name: samegenemodel, dtype: int64"
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mayra_pcrlist[mask_pcr_P]['samegenemodel'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    539\n",
       "0    353\n",
       "Name: samegenemodel, dtype: int64"
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mayra_pcrlist[mask_pcr_F]['samegenemodel'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Which means that samegenemodel category is high success rate for pcr. However, 539 genes that are in samegenemodel category need to be more investigated to figure out why they are not amplified. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "---\n",
    "Let's patch splited and merged column! \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17741"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import re\n",
    "def parse_cigar(scigar):\n",
    "        match = re.findall(r'(\\d+)(\\w)', scigar)\n",
    "        M = sum([int(x[0]) for x in match if x[1] == 'M'])\n",
    "        return M"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "100"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parse_cigar('99M197N1M')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
